File: libraries/analytics_debugger/src/main/java/com/tokopedia/analyticsdebugger/debugger/ui/adapter/FpmDebuggerTypeFactory.java
package com.tokopedia.analyticsdebugger.debugger.ui.adapter;

import android.view.View;

import com.tokopedia.abstraction.base.view.adapter.factory.BaseAdapterTypeFactory;
import com.tokopedia.abstraction.base.view.adapter.viewholders.AbstractViewHolder;
import com.tokopedia.analyticsdebugger.debugger.ui.model.FpmDebuggerViewModel;
import com.tokopedia.analyticsdebugger.debugger.ui.viewholder.FpmDebuggerViewHolder;

/**
 * @author okasurya on 5/16/18.
 */
public class FpmDebuggerTypeFactory extends BaseAdapterTypeFactory {

    @Override
    public AbstractViewHolder createViewHolder(View parent, int type) {
        if(type == FpmDebuggerViewHolder.LAYOUT) {
            return new FpmDebuggerViewHolder(parent);
        }

        return super.createViewHolder(parent, type);
    }

    public int type(FpmDebuggerViewModel fpmDebuggerViewModel) {
        return FpmDebuggerViewHolder.LAYOUT;
    }
}

File: libraries/analytics_debugger/src/main/java/com/tokopedia/analyticsdebugger/debugger/ui/adapter/FpmDebuggerTypeFactory.kt
package com.tokopedia.analyticsdebugger.debugger.ui.adapter

import android.view.View

import com.tokopedia.abstraction.base.view.adapter.factory.BaseAdapterTypeFactory
import com.tokopedia.abstraction.base.view.adapter.viewholders.AbstractViewHolder
import com.tokopedia.analyticsdebugger.debugger.ui.model.FpmDebuggerViewModel
import com.tokopedia.analyticsdebugger.debugger.ui.viewholder.FpmDebuggerViewHolder

/**
 * @author okasurya on 5/16/18.
 */
class FpmDebuggerTypeFactory : BaseAdapterTypeFactory() {

    override fun createViewHolder(parent: View, type: Int): AbstractViewHolder<*> {
        return if (type == FpmDebuggerViewHolder.LAYOUT) {
            FpmDebuggerViewHolder(parent)
        } else super.createViewHolder(parent, type)

    }

    fun type(fpmDebuggerViewModel: FpmDebuggerViewModel): Int {
        return FpmDebuggerViewHolder.LAYOUT
    }
}
